src/access_control.cairo

```diff
@@ -16,13 +16,14 @@ trait IAccessControl<TContractState> {
 #[starknet::component]
 mod access_control_component {
     use starknet::contract_address::ContractAddressZeroable;
+    use starknet::storage::Map;
     use starknet::{ContractAddress, get_caller_address};
 
     #[storage]
     struct Storage {
         admin: ContractAddress,
         pending_admin: ContractAddress,
-        roles: LegacyMap::<ContractAddress, u128>
+        roles: Map::<ContractAddress, u128>
```

Comment:
Is `Map` backwards compatible with `LegacyMap`? Asking because if someone uses access_control in an upgradable contract and then decides to upgrade both the lib and the contract, and if Map is not backwards compatible, they would loose all their setup, even brick the contract since they wouldn't be able to upgrade again.

Reply:
Good question. According to this [explainer](https://community.starknet.io/t/cairo-v2-7-0-is-coming/114362), the storage layout for `Map` is identical to `LegacyMap` so it should be backwards compatible.

Reply:
I managed to successfully test that an upgrade is successful in the following steps:
1. Deploy a Cairo `v2.6.5` contract using `v0.3.0` of this access control package ([class hash](https://sepolia.voyager.online/class/0x02179b4a9b5fffa59f06c7ecca5ed743f156cbb3f5967e0cbbc29378f06a2a60)). The contract has a setter function that asserts the caller has a role, and the deploying address is given the role at deployment.
2. Call the setter function as a sanity check.
3. Deploy a Cairo `v2.7.0` contract using this branch for the access control package ([class hash](https://sepolia.voyager.online/class/0x0645abdc9c5ee0c90dd07774c114a4954b085af00809161a1ee62ef31fa8206e)).
4. Call the setter function again. 

(4) was successful, meaning that the caller's role was correctly read from storage after the upgrade.

Reply:
Awesome! Thanks for the thorough check.